### PR TITLE
AP-1227 non-passported payload entity generation

### PR DESCRIPTION
**Pull request by stephenrichards**
## AP-1227 non-passported payload entity generation

[Link to story](https://dsdmoj.atlassian.net/browse/AP-1227)

- Created config files to control the generation of entities (a high-level container for attribute blocks in the payload) for passported and non-passported applications
- Added methods to the `NonPassportedCaseAddRequestor` which are referenced from the entity config files to determine whether or not the entity should be generated
- Tests for all the entity blocks 
- moved entity config files for passported and non-passported to their own subdirectory for clarity
- moved attribute block config files for passported and non-passported to their own subdirectory for clarity

## Checklist

Before you ask people to review this PR:

- [x] Tests and rubocop should be passing: `bundle exec rake`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- [x] The PR description should say what you changed and why, with a link to the JIRA story.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: config/ccms/means_entity_configs/non_passported.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  instance_label: the client's life assurance
+  yaml_section: :life_assurance
+
+- predicate: applicant_owns_share_of_property?
+  method: :generate_entity
+  entity_name: MAINTHIRD
+  instance_label: the third party who part owns the main dwelling1
+  yaml_section: :main_dwelling_third_party
+
+- predicate: true
```
Comment (colinbruce) [inline config/ccms/means_entity_configs/non_passported.yml]: This predicate doesn't match the others, should it?

File: spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_attributes_spec.rb [hunk 2]
```diff
@@ -40,10 +42,22 @@ module Requestors # rubocop:disable Metrics/ModuleLength
...
+            File.open(filename, 'w') { |f| f.puts xml }
+            expect(File.exist?(filename)).to be true
+          end
+        end
+
+        context 'hard coded false attributes', skip: 'skip until all enities have been coded up' do
           it 'generates the block with boolean value set to false' do
             false_attributes.each do |config_spec|
               entity, attribute, user_defined_ind = config_spec
+              puts ">>>>>>>>> #{entity} #{attribute} #{__FILE__}:#{__LINE__} <<<<<<<<<<\n"
```
Comment (colinbruce) [inline spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_attributes_spec.rb]: Debug statement left in

File: spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_entities_spec.rb [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+        let(:cfe_submission) { create :cfe_submission, legal_aid_application: legal_aid_application }
+        let!(:cfe_result) { create :cfe_result, submission: cfe_submission }
+        let(:requestor) { described_class.new(submission, {}) }
+        let(:xml) { requestor.formatted_xml }
+        let(:success_prospect) { :likely }
+        let(:merits_assessment) { create :merits_assessment, success_prospect: success_prospect, success_prospect_details: 'details' }
+
+        # enable this context if you need to create a file of the payload for manual inspection
+        # context 'saving to a temporary file', skip: 'Not needed for testing - but useful if you want to save the payload to a file' do
+        context 'save to a temporary file' do
```
Comment (colinbruce) [inline spec/services/ccms/requestors/case_add_requestor_xml_blocks/non_passported_entities_spec.rb]: Should this be set back to skip?

File: config/ccms/means_entity_configs/non_passported.yml (hunk 1 above)
Comment (stephenrichards) [reply config/ccms/means_entity_configs/non_passported.yml]: Yep - it just means always generate the entity block - don't rely on the result of a method to tell you whether or not to do it.

**Pull request merged**